lib/src/models/transaction_signing_mode.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/utils/constants.dart';
+
+part 'transaction_signing_mode.freezed.dart';
+
+@freezed
+sealed class TransactionSigningModes with _$TransactionSigningModes {
+   TransactionSigningModes._();
+
+   factory TransactionSigningModes.ordinaryTransaction() = OrdinaryTransaction;
+  factory TransactionSigningModes.multisigTransaction() = MultisigTransaction;
+  factory TransactionSigningModes.poolRegistrationAsOwner() = PoolRegistrationAsOwner;
+  factory TransactionSigningModes.poolRegistrationAsOperator() = PoolRegistrationAsOperator;
+  factory TransactionSigningModes.plutusTransaction() = PlutusTransaction;
+  
+  late final int value = switch (this) {
+    OrdinaryTransaction() => TransactionSigningMode.ordinaryTransaction.value,
+    MultisigTransaction() => TransactionSigningMode.multisigTransaction.value,
+    PoolRegistrationAsOwner() => TransactionSigningMode.poolRegistrationAsOwner.value,
+    PoolRegistrationAsOperator() => TransactionSigningMode.poolRegistrationAsOperator.value,
+    PlutusTransaction() => TransactionSigningMode.plutusTransaction.value,
+  };
```

Comment:
Ma gandeam recent ca ar fi bine sa facem asta peste tot (+ sa stergem enum-ul complet si sa bagam valoarea direct aici). Am putea de asemenea sa redenumim acest field din `value` in `serializeTypeValue` ca sa fie mai specific.

Dar cred ca e mai bine sa facem modificarea asta dupa ce adaugam integration tests.